vault/mount.go

```diff
@@ -303,6 +315,9 @@ func (c *Core) removeMountEntry(path string) error {
 
 // taintMountEntry is used to mark an entry in the mount table as tainted
 func (c *Core) taintMountEntry(path string) error {
+	c.mountsLock.Lock()
```

Comment:
If this is used to protect `persistMounts`, then this lock acquisition can be moved below the `SetTaint`.
Or if SetTaint *also* needs protection, then I guess `Tainted()` will need to use a read lock.

Also, can we do a `s/SetTaint/setTaint`?